convert.py

```diff
@@ -199,20 +202,20 @@ def guessed(model: LazyModel) -> Params:
     def loadHFTransformerJson(model: LazyModel, config_path: Path) -> Params:
         config = json.load(open(config_path))
 
-        n_vocab          = config["vocab_size"]
-        n_embd           = config["hidden_size"]
-        n_layer          = config["num_hidden_layers"]
-        n_ff             = config["intermediate_size"]
-        n_head           = config["num_attention_heads"]
-        n_head_kv        = config["num_key_value_heads"] if "num_key_value_heads" in config else n_head
-        f_norm_eps       = config["rms_norm_eps"]
-        f_rope_freq_base = config["rope_theta"] if "rope_theta" in config else None
-
+        rope_scaling_type = f_rope_scale = n_orig_ctx = rope_finetuned = None
         rope_scaling = config.get("rope_scaling")
-        if isinstance(rope_scaling, dict) and rope_scaling.get("type") == "linear":
-            f_rope_scale = config["rope_scaling"].get("factor")
-        else:
-            f_rope_scale = None
+
+        if rope_scaling is not None and (typ := rope_scaling.get("type")):
+            rope_factor = rope_scaling.get("factor")
+            f_rope_scale = rope_factor
+            if typ == "linear":
+                rope_scaling_type = gguf.RopeScalingType.LINEAR
+            elif typ == "yarn":
+                rope_scaling_type = gguf.RopeScalingType.YARN
+                n_orig_ctx = rope_scaling['original_max_position_embeddings']
+                rope_finetuned = rope_scaling['finetuned']
```

Comment:
conceptofmind's models do not contain this field.
https://huggingface.co/conceptofmind/Yarn-Llama-2-7b-128k/blob/main/config.json

Reply:
assume true if `original_max_position_embeddings` exists?

Reply:
But in theory, someone could create a non-finetuned model that uses some variant of RoPE scaling by default. Maybe instead of "no" the status should be "unknown"?

Reply:
> non-finetuned model that uses some variant of RoPE scaling by default. 

but would then there be an "original" ctx size ?

Reply:
The original context size would be whatever the model usually uses, and the new context size would be the maximum it chooses to support via YaRN. Finetuning is not mandatory, YaRN is the best context extension method either way.

Reply:
ok, then why do we look at the value anyway?

Reply:
I think it's useful information for the user (it's printed when the model is loaded), since finetuned models perform better, but a model doesn't have to be finetuned to specify YaRN.

Reply:
ah, yea sure, i thought it was used for a calculation.